src/coins.cpp

```diff
@@ -503,13 +512,47 @@ void CCoinsViewCache::SetBestBlock(const uint256 &hashBlockIn) {
     hashBlock = hashBlockIn;
 }
 
+void CCoinsViewCache::AddCswNullifier(const uint256& scId,
+                         const libzendoomc::ScFieldElement &nullifier) {
+    std::pair<uint256, libzendoomc::ScFieldElement> position = std::make_pair(scId, nullifier);
+    std::pair<CCswNullifiersMap::iterator, bool> ret = cacheCswNullifiers.insert(std::make_pair(position, CCswNullifiersCacheEntry()));
+    ret.first->second.entered = true;
+    ret.first->second.flags |= CCswNullifiersCacheEntry::FRESH;
+}
+
+void CCoinsViewCache::RemoveCswNullifier(const uint256& scId,
+                         const libzendoomc::ScFieldElement &nullifier) {
+    std::pair<uint256, libzendoomc::ScFieldElement> position = std::make_pair(scId, nullifier);
+    std::pair<CCswNullifiersMap::iterator, bool> ret = cacheCswNullifiers.insert(std::make_pair(position, CCswNullifiersCacheEntry()));
+    ret.first->second.entered = false;
+    ret.first->second.flags |= CCswNullifiersCacheEntry::DIRTY;
+}
+
+bool CCoinsViewCache::GetCswNullifier(const uint256& scId,
+                         const libzendoomc::ScFieldElement &nullifier) const {
+    std::pair<uint256, libzendoomc::ScFieldElement> position = std::make_pair(scId, nullifier);
+    
+    CCswNullifiersMap::iterator it = cacheCswNullifiers.find(position);
+    if (it != cacheCswNullifiers.end())
+        return true;
```

Comment:
instead of `return true` -> check the `it` flag value.
It may be the case, when in `base` the nullifer was present, but in the `cache` we removed it. So, cache should return `false`, because iterator `flag == ERASED`.